frontend/src/scenes/organization/Settings/DangerZone.tsx

```diff
@@ -28,15 +29,32 @@ export function DeleteOrganizationModal({
                 // @ts-expect-error - data-attr works just fine despite not being in ButtonProps
                 'data-attr': 'delete-organization-ok',
                 loading: isDeletionInProgress,
+                disabled: !isDeletionConfirmed,
             }}
             onCancel={() => setIsVisible(false)}
             cancelButtonProps={{
                 disabled: isDeletionInProgress,
             }}
             visible={isVisible}
         >
-            Organization deletion <b>cannot be undone</b>. You will lose all data, <b>including all events</b>, related
-            to all projects within this organization.
+            <p>
+                Organization deletion <b>cannot be undone</b>. You will lose all data, <b>including all events</b>,
+                related to all projects within this organization.
+            </p>
+            <p>
+                Please type{' '}
+                <strong>{currentOrganization ? currentOrganization.name : "this organization's name"}</strong> to
+                confirm.
+            </p>
+            <Input
+                type="text"
+                onChange={(e) => {
+                    if (currentOrganization) {
+                        const { value } = e.target
+                        setIsDeletionConfirmed(value.toLowerCase() === currentOrganization.name.toLowerCase())
```

Comment:
Same here, if the `currentOrganization` = `null`, wouldn't this always fail? Or would it never be `null`? if it's never `null`, why the need for the line 46.

Reply:
Well, if it's actually `null` then you wouldn't be seeing this screen, since there'd no organization (or project in the other case) to delete. It's `null` only for a short time when loading from the API, but we can be sure that that has already happen if you're viewing the settings page. If it's null otherwise, that'd mean you belong to no organization and you'd be redirected to the org creation page. :)